src/drivers/uefi/lv_uefi_display.c

```diff
@@ -0,0 +1,284 @@
+    #error [lv_uefi] Unsupported LV_COLOR_DEPTH.
+#endif
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+typedef struct _lv_uefi_display_context_t {
+    EFI_HANDLE handle;
+    EFI_GRAPHICS_OUTPUT_PROTOCOL * gop_protocol;
+    void * buffer;
+    size_t buffer_size;
+} lv_uefi_display_context_t;
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+static void _display_event_cb(lv_event_t * e);
+static void _display_flush_cb(lv_display_t * display, const lv_area_t * area, uint8_t * px_map);
+
+static void _display_ctx_free(lv_uefi_display_context_t * display_ctx);
+static bool _display_interface_is_valid(const EFI_GRAPHICS_OUTPUT_PROTOCOL * interface);
+
+/**********************
+ *  GOLBAL VARIABLES
+ **********************/
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+static EFI_GUID _uefi_guid_graphics_output = EFI_GRAPHICS_OUTPUT_PROTOCOL_GUID;
+static EFI_GUID _uefi_guid_edid_active = EFI_EDID_ACTIVE_PROTOCOL_GUID;
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+/**
+ * @brief Create a LVGL display object.
+ * @param handle The handle on which an instance of the EFI_GRAPHICS_OUTPUT_PROTOCOL protocol is installed.
+ * @return The created LVGL display object.
+ */
+lv_display_t * lv_uefi_display_create(void * handle)
+{
+    lv_display_t * display = NULL;
+    lv_uefi_display_context_t * display_ctx;
+
+    if(!lv_uefi_protocol_test(handle, &_uefi_guid_graphics_output)) return NULL;
+
+    display_ctx = lv_calloc(1, sizeof(lv_uefi_display_context_t));
+    LV_ASSERT_MALLOC(display_ctx);
+
+    display_ctx->handle = handle;
+    display_ctx->gop_protocol = (EFI_GRAPHICS_OUTPUT_PROTOCOL *)lv_uefi_protocol_open(handle, &_uefi_guid_graphics_output);
+    if(!_display_interface_is_valid(display_ctx->gop_protocol)) {
+        LV_LOG_WARN("[lv_uefi] The GOP interface is not valid.");
+        goto error;
+    }
+
+    // 4 bytes per pixel
+    display_ctx->buffer_size = 4 * display_ctx->gop_protocol->Mode->Info->HorizontalResolution *
+                               display_ctx->gop_protocol->Mode->Info->VerticalResolution;
+    display_ctx->buffer = lv_malloc(display_ctx->buffer_size);
+    LV_ASSERT_MALLOC(display_ctx->buffer);
+
+    display = lv_display_create(display_ctx->gop_protocol->Mode->Info->HorizontalResolution,
+                                display_ctx->gop_protocol->Mode->Info->VerticalResolution);
+    lv_display_add_event_cb(display, _display_event_cb, LV_EVENT_DELETE, display);
+    lv_display_set_flush_cb(display, _display_flush_cb);
+    lv_display_set_buffers(display, display_ctx->buffer, NULL, (uint32_t)display_ctx->buffer_size,
+                           LV_DISPLAY_RENDER_MODE_DIRECT);
+    lv_display_set_user_data(display, display_ctx);
+
+    goto finish;
```

Comment:
Any specific reason to use label instead of early return if everything went well?

Reply:
Again personal preference, I like to have a dedicated place to free allocated ressources and handle errores inside a function. But I can change that if it does not play nice with the LVGL coding style.